urllib3/response.py

```diff
@@ -117,8 +118,17 @@ def __init__(self, body='', headers=None, status=0, version=0, reason=None,
         if hasattr(body, 'read'):
             self._fp = body
 
-        if preload_content and not self._body:
-            self._body = self.read(decode_content=decode_content)
+        # are we using the chunked-style of transfer encoding?
+        tr_enc = self.headers.get('transfer-encoding')
+        if tr_enc and tr_enc.lower() == "chunked":
+            self.chunked = True
+            self.chunk_left = None
+        else:
```

Comment:
okay, makes sense to me (I tried to copy most of the code from httplib; my reasoning was that if it's in httplib, it gotta be perfect) — will update